### ISSUE TITLE
can't switch to gpt-4o-mini

### ISSUE THREAD

**Issue by rugene76** [jekalmin/extended_openai_conversation#244]
I can't get the integration to switch to gpt-4o-mini model.

I enter "gpt-4o-mini" into the field in the configure dialog, reload and even reboot HA but the chatgpt reports that its using the 3.5 architecture.

What am I missing?

thanks

**Comment by rugene76** [comment]
how the heck did you figure that out? !!
That worked! yea!  On to the next issue.
btw, I've put together both an esp32 and raspi w2 AI hardware, after much tweaking none of them are working well enough to replace my google hubs. Too slow to respond, unreliable wakeword, false wakeword detections....frustrating.

**Comment by kirel** [comment]
@rugene76 try procupine1 instead of open-wakeword. For me that made all the difference.

**Comment by Al3XKOoL** [comment]
I have working procupine3 in my repo https://github.com/Al3XKOoL/hassio-addons

Its working for me and custom wake words are working.
Advantages of v3 compared to v1:

Improved accuracy
More supported languages and pre-trained wake words
Ability to use custom wake words
Drawbacks:

An access token from console.picovoice.ai is needed
Each token can be used on only 3 machines per month

**Comment by oywino** [comment]
> I also had this problem but had to change it to "gpt-4o-mini-2024-07-18". Try that!

That didn't work! It still uses 3.5

**Comment by jekalmin** [comment]
Please attach error logs if the problem still exists.

**Comment by jekalmin** [comment]
No need to restart HA whenever you change configuration.
There should be logs in your `/config/home-assistant.log`.
If you use HAOS, you can access it via `Terminal & SSH` addon, and type below.

```sh
tail -f /root/config/home-assistant.log
```

**Comment by oywino** [comment]
So, I issued the `tail `command and then changed the chat-model setting in the config to `gpt-4o-mini-2024-07-18` but there was no log activity what so ever.

**Comment by jekalmin** [comment]
Did you try assist too while tailing logs?

**Comment by oywino** [comment]
Yes, and now I did it again, testing both using Assist through my web Browser as well as using the HA Voice PE.
The log is dead silent.  Both attempts works and my command given is carried out perfectly what I ask it switch some light sources.
But when I ask for the name of the president in USA, the answer is Jo Biden 😮
And when I ask it what GPT version it is, the answer is 3.5

**Comment by dotorg** [comment]
So, just FYI, 4o-mini *would* say Joe Biden. It's from July 2024 and none of the gpt-x models have access to the Internet, so they only know what they were trained on. For newer data, you have to provide functions that can run appropriate queries and return either the answers or data the model can parse for the answer.  That's just how LLMs work. ChatGPT is OpenAI's product built on gpt-4o with those functions built in, which is why it can answer current event questions (at least sometimes).

**Comment by oywino** [comment]
Well, that's ok, just ignore it. 
But why does it still "work" when I set chat_model = Santa Claus ?

**Comment by dotorg** [comment]
A quick skim of the code, it looks like the schema definition has the setting as "chat_model" but the services.py code is just "model" with a default of "gpt-4-vision-preview" if no value is set, but the default in the setup path is "gpt-4o-mini".

That default was changed in this commit: https://github.com/jekalmin/extended_openai_conversation/commit/94d293f919ae769f0494fc5c598f3ed5b0a80130

So my guess would be that you're running a version older than that commit and the setting has never worked in the UI.

**Comment by oywino** [comment]
Well, mine says 1.0.4 and the one you're referring to is 1.0.5 Beta2, right?
I'm not sure how to install that version. Can't find it in HACS. Perhaps I should wait until you're ready to release 1.0.5?
